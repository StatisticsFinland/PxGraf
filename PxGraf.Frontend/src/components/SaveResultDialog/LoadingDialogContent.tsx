import { CircularProgress, Container, DialogContent } from '@mui/material';
import React from 'react';
import styled from 'styled-components';

const ProgressWrapper = styled(Container)`
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const LoadingDialogContent: React.FC = () => {
    return (
        <DialogContent dividers={true}>
            <ProgressWrapper>
                <CircularProgress />
            </ProgressWrapper>
        </DialogContent>
    );
}

export default LoadingDialogContent;