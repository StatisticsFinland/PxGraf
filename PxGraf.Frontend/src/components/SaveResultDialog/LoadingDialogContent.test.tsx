import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LoadingDialogContent } from './LoadingDialogContent';

describe('Rendering test', () => {
    it('renders correctly', () => {
        // for some
        const { asFragment } = render(<LoadingDialogContent />);
        expect(asFragment()).toMatchSnapshot();
    });
});