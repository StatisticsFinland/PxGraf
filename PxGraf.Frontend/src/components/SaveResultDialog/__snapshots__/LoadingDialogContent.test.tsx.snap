// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiDialogContent-root MuiDialogContent-dividers css-10bxh4q-MuiDialogContent-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg sc-Qotzb bfUJAt css-5c1adp-MuiContainer-root"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
